### ISSUE TITLE
Untitled workspace is always remembered even when hot-exit is off 

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#31026]
Testing #30722:

- disable hot-exit
- Create an untitled workspace with two folders
- close workbench window

Observe: you are not asked whether you want to save the workspace but when restarting VS Code the untitled workspace is opened automatically

**Comment by bpasero** [comment]
@weinand I am not sure we should correlate untitled workspaces with the `files.hotExit` setting. 

There should however be no different behaviour with untitled workspaces depending on the `files.hotExit` setting, so I wonder if you saw a difference? That would indicate there is a bug.

**Comment by weinand** [comment]
In the test item there was this sentence:
> you get asked to save an untitled workspace **when you close the window** or load something else into the window

Since I was never asked to save the untitled workspace when closing the window, I was assuming that hot-exit took place. But turning this off did not make a difference.

If this is the desired behaviour, I conclude that it is impossible to be asked for saving the untitled workspace when closing the window, correct? (if yes, then we should fix the sentence in the test plan).

**Comment by bpasero** [comment]
I have updated the test plan with this:

> you get asked to save an untitled workspace when you load something else into the window or when you open a second window and close the first window with the untitled workspace

**Comment by weinand** [comment]
> open a second window and close the first window with the untitled workspace
aha, I didn't know that... now I have to retest everything ;-)

**Comment by bpasero** [comment]
@weinand the thing is, if you close the last window (on Windows/Linux at least) there is a certain semantic to it that makes it different from closing a window: The intent of the user is most likely to exit the application, not necessarily to get rid of the window (as opposed to macOS where you can close a window and the app is still running). So in that case we do not ask to save the workspace but simply restore it. The same happens when you open multiple windows with untitled workspaces and do File > Quit.

Basically the rule is this:
* user quits: do not ask to save workspace
* user closes a window: ask to save workspace (unless the last window is closed on Windows/Linux)
